packages/matter.js/src/protocol/interaction/InteractionServer.ts

```diff
@@ -1126,156 +1150,257 @@ export class InteractionServer implements ProtocolHandler<MatterDevice>, Interac
-                            if (error instanceof ValidationError) {
-                                logger.info(
-                                    `Validation-${errorLogText}${error.fieldName !== undefined ? ` in field ${error.fieldName}` : ""}`,
-                                );
-                            } else {
-                                logger.info(errorLogText);
-                            }
-                            return {
-                                code: error.code,
-                                clusterCode: error.clusterCode,
-                                responseId: command.responseId,
-                                response: TlvNoResponse.encodeTlv(),
-                            };
-                        },
-                    );
-                    const { code, clusterCode, responseId, response } = result;
-                    if (response.length === 0) {
-                        invokeResponses.push({
-                            status: { commandPath: path, status: { status: code, clusterStatus: clusterCode } },
-                        });
-                    } else {
-                        invokeResponses.push({
-                            command: {
-                                commandPath: { ...path, commandId: responseId },
-                                commandFields: response,
-                            },
+                    continue;
+                }
+                if (
+                    getMatterModelClusterCommand(clusterId, commandId)?.fabricScoped &&
+                    (!exchange.session.isSecure || !(exchange.session as SecureSession<MatterDevice>).fabric)
+                ) {
+                    logger.debug(`This invoke requires a secure session with a fabric assigned which is missing.`);
+                    if (isConcretePath) {
+                        await processResponseResult({
+                            status: { commandPath: path, status: { status: StatusCode.UnsupportedAccess }, commandRef },
                         });
                     }
+                    continue;
                 }
-            }),
-        );
 
-        // TODO support suppressResponse for responses
-        return {
-            suppressResponse: false,
-            interactionModelRevision: INTERACTION_MODEL_REVISION,
-            invokeResponses,
-        };
+                const result = await tryCatchAsync(
+                    async () =>
+                        await this.invokeCommand(
+                            path,
+                            command,
+                            exchange,
+                            commandFields ?? TlvNoArguments.encodeTlv(commandFields),
+                            message,
+                            endpoint,
+                            receivedWithinTimedInteraction,
+                        ),
+                    StatusResponseError,
+                    async error => {
```

Comment:
And the error handler parameter to tryCatchAsync seems like a good candidate for MaybeAsync given this isn't actually async.